src/transformations.jl

```diff
@@ -97,22 +121,69 @@ julia> StatsBase.transform(dt, X)
  -1.0   0.0  1.0
 ```
 """
-function fit(::Type{ZScoreTransform}, X::AbstractArray{<:Real,2}; center::Bool=true, scale::Bool=true)
-    d, n = size(X)
-    n >= 2 || error("X must contain at least two columns.")
+function fit(::Type{ZScoreTransform}, X::AbstractMatrix{<:Real};
+             dims::Union{Integer,Nothing}=nothing, center::Bool=true, scale::Bool=true)
+    if dims == 1
+        n, l = size(X)
+        n >= 2 || error("X must contain at least two rows.")
+        m, s = mean_and_std(X, 1)
+    elseif dims == 2
+        l, n = size(X)
+        n >= 2 || error("X must contain at least two columns.")
+        m, s = mean_and_std(X, 2)
+    elseif dims === nothing
+        Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
```

Comment:
Need to do the same thing as when `dims==2`. Same below.

Also need to throw an error if `dims` takes another value.

Reply:
Hmmmmmm, I need to add `depwarn` when `dims==2`?
If yes, what for?

Reply:
I said "the same thing as when", not "the same thing when". I mean that you need to set `m` and `s`.

Reply:
Oh, I got it.